xlators/cluster/afr/src/afr-common.c

```diff
@@ -2587,12 +2576,11 @@ afr_local_cleanup(afr_local_t *local, xlator_t *this)
 
     if (!local)
         return;
+    priv = this->private;
 
     syncbarrier_destroy(&local->barrier);
 
-    afr_local_transaction_cleanup(local, this);
-
-    priv = this->private;
+    afr_local_transaction_cleanup(local, priv);
```

Comment:
Sending priv->child_count should suffice here.
```suggestion
    afr_local_transaction_cleanup(local, priv->child_count);
```